pkg/controller/dnszone/dnszone_controller.go

```diff
@@ -152,6 +154,19 @@ func (r *ReconcileDNSZone) Reconcile(ctx context.Context, request reconcile.Requ
 	}
 
 	if result, err := controllerutils.ReconcileDNSZoneForRelocation(r.Client, dnsLog, desiredState, hivev1.FinalizerDNSZone); err != nil {
+		var changed bool
+		desiredState.Status.Conditions, changed = controllerutils.SetDNSZoneConditionWithChangeCheck(
+			desiredState.Status.Conditions,
+			hivev1.GenericDNSErrorsCondition,
+			corev1.ConditionTrue,
+			"RelocationError",
+			controllerutils.ErrorScrub(err),
+			controllerutils.UpdateConditionIfReasonOrMessageChange)
+		if changed {
+			if err := r.Status().Update(context.Background(), desiredState); err != nil {
+				dnsLog.WithError(err).Log(controllerutils.LogLevel(err), "failed to update dnszone status")
+			}
+		}
```

Comment:
Would this get cleared if it was set, and then the problem resolved?

Reply:
Yes, it will be [here](https://github.com/openshift/hive/pull/1432/files#diff-3054a0cfe6b1c7c844ae261bfb40f5ad58f26b7c4615079cf3dbff0b034e6c4fL217). I made some changes and also ensured there's a [test-case](https://github.com/openshift/hive/pull/1432/files#diff-ae52ca54b986b416ec91f5fe092abedf45e6fd73b413fc92791620e459fd2a91R540-R564) for this scenario